Mathlib/Topology/DiscreteSubset.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash, Bhavik Mehta
+-/
+import Mathlib.Topology.SubsetProperties
+import Mathlib.Topology.Separation
+
+/-!
+# Discrete subsets of topological spaces
+
+Given a topological space `X` together with a subset `s ⊆ X`, there are two distinct concepts of
+"discreteness" which may hold. These are:
+  (i) Every point of `s` is isolated (i.e., the subset topology induced on `s` is the discrete
+      topology).
+ (ii) Every compact subset of `X` meets `s` only finitely often (i.e., the inclusion map `s → X`
+      tends to the cocompact filter along the cofinite filter on `s`).
+
+When `s` is closed, the two conditions are equivalent provided `X` is locally compact and T1,
+see `IsClosed.tendsto_coe_cofinite_iff`.
+
+## Main definitions
+
+ * `tendsto_cofinite_cocompact_iff`:
+ * `IsClosed.tendsto_coe_cofinite_iff`:
+
+-/
+
+open Set Filter Function Topology
+
+variable {X Y : Type _} [TopologicalSpace X] [TopologicalSpace Y] {f : X → Y}
+
+lemma tendsto_cofinite_cocompact_iff :
+    Tendsto f cofinite (cocompact _) ↔ ∀ K, IsCompact K → Set.Finite (f ⁻¹' K) := by
```

Comment:
I think that `hasBasis_cocompact` can help golf this proof. Something like (not tested):
```
  rw [hasBasis_cocompact.tendsto_right_iff, ← compl_surjective.forall]
  simp only [compl_compl, preimage_compl]
```

Reply:
Thanks, I've pushed a proof using `hasBasis_cocompact.tendsto_right_iff` (the above didn't quite work).